ViberApiLib.Test/Test.cs

```diff
@@ -0,0 +1,305 @@
+﻿using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+
+namespace ViberApiLib.Test
+{
+    [TestFixture]
+    public class Test
+    {
+        private Api api;
+
+        [SetUp]
+        protected void SetUp()
+        {
+            api = new Api("testAuthToken", "testBotName", "testAvatarPath");
+        }
+
+        [Test]
+        public void TestSubscribedRequest()
+        {
+            var reqData = @"{
+               ""event"":""subscribed"",
+               ""timestamp"":1457764197627,
+               ""user"":{
+                  ""id"":""01234567890A="",
+                  ""name"":""John McClane"",
+                  ""avatar"":""http://avatar.example.com"",
+                  ""country"":""UK"",
+                  ""language"":""en"",
+                  ""api_version"":1
+               },
+               ""message_token"":4912661846655238145
+            }";
+            var req = api.ParseRequest(reqData.ToString());
+            Assert.IsInstanceOf<SubscribedRequest>(req);
+            Assert.AreEqual(req.Event, "subscribed");
+            Assert.AreEqual(req.TimeStamp, "1457764197627");
+            if (req is SubscribedRequest)
+            {
+                SubscribedRequest dreq = req as SubscribedRequest;
```

Comment:
This is spec that consumer do "down casting" because the Viber API's Request model doesn't have interface (each Request have different element).

Reply:
I see.